app/api/v1/schedule.py

```
@@ -0,0 +1,96 @@
+                description=description,
+                location=location,
+            )
+            await schedule_service.update_current_schedule(
+                agent_id=agent_config.id,
+                schedule_item=schedule_item,
+            )
+            return schedule_item
+        else:
+            raise HTTPException(status_code=404, detail="Schedule not found")
+    except KeyError:
+        raise HTTPException(status_code=404, detail="Agent not found")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)